gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/grouping/GroupingGroupCombine.java

```diff
@@ -129,9 +145,35 @@ protected LG groupInternal(LG graph) {
       .map(new BuildVertexWithSuperVertexBC())
       .withBroadcastSet(mapping, BuildVertexWithSuperVertexBC.BC_MAPPING);
 
-    // build super edges
-    DataSet<E> superEdges = buildSuperEdges(graph, vertexToRepresentativeMap);
+    DataSet<E> edgesToGroup = graph.getEdges();
+
+    if (optionalRetainedVerticesSubgraph.isPresent()) {
```

Comment:
```
if (isRetainingVerticesWithoutGroups()) {
      LG retainedVerticesSubgraph = getSubgraphOfRetainedVertices(graph);

      // To add support for grouped edges between retained vertices and supervertices,
      // vertices are their group representatives themselves
      vertexToRepresentativeMap =
        updateVertexRepresentatives(vertexToRepresentativeMap,
          retainedVerticesSubgraph.getVertices());

      // don't execute grouping on edges between retained vertices
      // but execute on edges between retained vertices and grouped vertices
      //   graph.getEdges() - retainedVerticesSubgraph.getEdges()
      edgesToGroup = subtractEdges(graph.getEdges(), retainedVerticesSubgraph.getEdges());
    }

    DataSet<E> superEdges =
      buildSuperEdges(graph.getFactory().getEdgeFactory(), edgesToGroup, vertexToRepresentativeMap);

    if (isRetainingVerticesWithoutGroups()) {

      LG retainedVerticesSubgraph = getSubgraphOfRetainedVertices(graph);
      superVertices = superVertices.union(retainedVerticesSubgraph.getVertices());
      superEdges = superEdges.union(retainedVerticesSubgraph.getEdges());
    }
```
Instead of this whole block.